### PR TITLE
vim-patch: runtime updates

### LINKED ISSUE #7502: Recognize NeoMutt in `filetype.vim`
NeoMutt is a fork of Mutt with added features (a strict superset). It used to install as `mutt`, but recently it changed name to `neomutt` in order to allow side-by-side installation with "regular" Mutt — and now when composing mails, the `neomutt` prefix for the temporary file is no longer recognized as the old `mutt` prefix is.

Vim was already changed to recognize NeoMutt: https://github.com/vim/vim/commit/01164a6546b4c635daf96a1f17d1cb2d07f32a66

Reference: https://github.com/neomutt/neomutt/issues/927
There's a vim plugin as well: https://github.com/neomutt/neomutt.vim

However, it would still be nice if https://github.com/neovim/neovim/blob/52748d266ddbb73e0c759a5251ecbeda1c5fcade/runtime/filetype.vim#L1151 was amended to recognize the `neomutt` prefix.

### PR DESCRIPTION
**Pull request by justinmk**
- Note: Ignored changes to matchit.vim in favor of faca81411628.
- Note: there are a small number of minor doc changes that require vim patches we haven't merged yet. Consider it a minor bug.
- Removed `runtime/vimrc_example.vim`, it's mostly irrelevant. Any apology for it should be redirected to a built-in default.

closes #7502

### CODE REVIEW

File: scripts/vim-patch.sh [hunk 1]
```diff
@@ -226,7 +226,7 @@ stage_patch() {
       printf "\n✘ 'patch' command not found\n"
     else
       printf "\nApplying patch...\n"
-      patch -p1 < "${patch_file}"
+      patch -p1 --no-backup-if-mismatch < "${patch_file}"
```
Comment (jamessan) [inline scripts/vim-patch.sh]: `--no-backup-if-mismatch` is a GNU-ism.  `--posix` causes the same behavior, but also has some other effects.

**Comment by jamessan** [comment]
The final two commits can be squashed.  The first is the empty commit from running vim-patch.sh, but the second one has the content.

File: runtime/optwin.vim [hunk 2]
```diff
@@ -1164,6 +1164,8 @@ endif
 call append("$", "iminsert\tin Insert mode: 1: use :lmap; 2: use IM; 0: neither")
 call append("$", "\t(local to window)")
 call <SID>OptionL("imi")
+call append("$", "imstyle\tinput method style, 0: on-the-spot, 1: over-the-spot")
```
Comment (jamessan) [inline runtime/optwin.vim]: This option doesn't exist in nvim.

Comment (justinmk) [reply runtime/optwin.vim]: Yeah, I was wondering if I should bother. Not sure who actually uses optwin.vim.

I'll change it since you mentioned it.

Comment (jamessan) [reply runtime/optwin.vim]: I only mentioned it because of the failed tests. :)

**Comment by justinmk** [comment]
travis failure https://github.com/neovim/neovim/issues/7504 seems correlated with #7500 but could just be travis nonsense.

**Pull request merged**